server/pulp/server/async/tasks.py

```diff
@@ -780,11 +782,7 @@ def _handle_cProfile(self, task_id):
         if config.getboolean('profiling', 'enabled') is True:
             self.pr.disable()
             profile_directory = config.get('profiling', 'directory')
-            try:
-                os.makedirs(profile_directory, 0755)
-            except OSError as exc:
-                if exc.errno != errno.EEXIST:
-                    raise
+            misc.mkdir(profile_directory, mode=0755)
```

Comment:
0775

Reply:
i think this is important and i understand correctly how umask works - if having mode 0755 the result will be:
 usr grp others 
-rwx  r-x   r-x is represented in octal as 0755
-000 000 010 umask of 002 removes any permission where there is a 1
-rwx  r-x   r-x is the result achieved requesting 0755 with umask of 002...and we are screwed :)


if having mode 0775 the result will be:
 usr grp others 
-rwx  rwx  r-x is represented in octal as 0775
-000 000 010 umask of 002 removes any permission where there is a 1
-rwx  rwx  r-x   is the result achieved requesting 0775 with umask of 002

Reply:
I left a few things like certificate-directories and this profiling-directory alone - they're not 'content', and are more sensitive to permissions. Unless we're migrating profiling-information, this set of perms shouldn't affect migration, and I felt it better to leave the permissions as they had been specifically set.

(Your understanding of umask is correct - it turns off permission-bits where the umask has a 1. So 0755, with the mkdirs() umask of 0002, leaves 0755)